src/config.c

```diff
@@ -682,14 +840,32 @@ int ReadConfig(RedisModuleString **argv, int argc, char **err) {
       return REDISMODULE_ERR;
     }
 
-    // `triggerId` is set by the coordinator when it registers a trigger for a configuration.
-    // If we don't have a coordinator or this configuration has no trigger, this value
-    // is meaningless and should be ignored
-    if (curVar->setValue(&RSGlobalConfig, &ac, curVar->triggerId, &status) != REDISMODULE_OK) {
-      *err = rm_strdup(QueryError_GetError(&status));
-      QueryError_ClearError(&status);
-      return REDISMODULE_ERR;
+    // if the value is different from default, we can't update it, because the
+    // CONFIG SET has higher priority than the FT.CONFIG SET
+    RSConfig RSDefaultConfig = RS_DEFAULT_CONFIG;
+    sds currValue = curVar->getValue(&RSGlobalConfig);
+    sds currValueDefault = curVar->getValue(&RSDefaultConfig);
+
+    if ((currValue != NULL && currValueDefault!= NULL
+          && sdscmp(currValue, currValueDefault) == 0 ) ||
+          (currValue == NULL && currValueDefault == NULL)) {
+      // `triggerId` is set by the coordinator when it registers a trigger for a configuration.
+      // If we don't have a coordinator or this configuration has no trigger, this value
+      // is meaningless and should be ignored
+      if (curVar->setValue(&RSGlobalConfig, &ac, curVar->triggerId, &status) != REDISMODULE_OK) {
+        *err = rm_strdup(QueryError_GetError(&status));
+        QueryError_ClearError(&status);
+        return REDISMODULE_ERR;
+      }
+    } else {
+      // Consume the value
+      RedisModule_Log(NULL, "notice", "Can't set %s, because it was already set by CONFIG", curVar->name);
```

Comment:
Don't we want to return an error and fill err with this string instead?
This logic seems a bit off.